### DISCUSSION TITLE
Multivariate time series clustering with variable-lengths

### DISCUSSION THREAD

**Discussion by Bitaom** [tslearn-team/tslearn#423] [Q&A]
I have been trying to use the times series kmeans algorithm for clustering time series with 4 dimensions and variable lengths. According to the documentation, TimeseriesKMeans can handle variable lengths. In my dataset, shorter time series have "Nan" values for the time periods that there is no data available. When I pass the data set through the model it is not able to cluster and just returns one cluster including all of the time series. On the other hand, when I replace the "Nan" values with zeros I get reasonable clustering results. But in my case, it is not logical to add zeros to the shorter series. 
Is the algorithm capable of dealing with nan values? Should I make any changes to the dataset format? I appreciate any recommendations for this issue.

**Comment by jahchwe** [comment]
Hi Bitaom, 

I am currently dealing with a similar challenge (multivariate timeseries with variable lengths). I thought that using dynamic time warping before clustering the timeseries could be a solution. 

How did you end up moving forward on this? 

Thanks!
